### ISSUE TITLE
Refactor LCD endpoints

### ISSUE THREAD

**Issue by faboweb** [cosmos/voyager#1010]
Description:

We want to improve the LCD staking endpoints:
- to be more REST-ful
- to allow batch querying of stake per delegator

**Comment by faboweb** [comment]
Consider #879

**Comment by fedekunze** [comment]
I can work on this too @faboweb

**Comment by faboweb** [comment]
Current state:
query delegation `/stake/{delegator}/delegation/{validator}`
query undonding delegations `/stake/{delegator}/ubd/{validator}`
query redelegations `/stake/{delegator}/red/{validator_src}/{validator_dst}`
query candidates `/stake/validators`
post bonding/unbdonding/redelegation tx `/stake/delegations`

**Comment by fedekunze** [comment]
- [ ] Change endpoints on the SDK
- [ ] Write tests on the SDK
- [ ] Implement on Voyager
- [ ] Write tests on Voyager
- [ ] Change LCD client yaml

**Comment by faboweb** [comment]
Stretch goal could be implementing this in Voyager (should be handled in a new issue)

**Comment by fedekunze** [comment]
@faboweb Just had a quick talk with Ismail and we agreed on taking out the `/stake` word from the beginning of the endpoint. Please also check this [article](https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9) for API best practices

**Comment by NodeGuy** [comment]
I read that article and agree with it.

**Comment by faboweb** [comment]
@fedekunze 

> we agreed on taking out the /stake

Can you add some reasoning?

**Comment by fedekunze** [comment]
@faboweb because it's not actually needed and it doesn't add much value to the endpoint (as the article says). If we want to get the module an endpoint is coming from we could for example add `Tag`s in the `Result`s with `module` equal to `stake` (as in this case)

**Comment by faboweb** [comment]
This is not about mapping the results but about namespacing the endpoints coming from different modules (`stake` module, `gov` module) to not conflict endpoints. I read the article and it doesn't handle this case. Please contact the SDK team if you want to change this convention.
One more thought: Requesting data and only after receiving it getting to know what data I requested sounds impractical.

**Comment by cwgoes** [comment]
>  I am fine with having no namespacing, but let's contact the SDK team if we want to change this convention (pulling in @cwgoes ).

Hmm I understand the desire to simplify but I think this will lead to problems. We want standard REST endpoints for not only our module set within Gaia but any conceivable other SDK apps which use our standard modules - those apps may use their own modules too, which might have conflicting endpoint names. If we prefix all routes by the name of the module, downstream usage is much easier.

**Comment by faboweb** [comment]
I will close this issue, as this has been merged on the SDK side.
